src/Unleash/DefaultUnleash.cs

```diff
@@ -248,108 +100,17 @@ public Variant GetVariant(string toggleName, UnleashContext context)
 
         public Variant GetVariant(string toggleName, UnleashContext context, Variant defaultValue)
         {
-            var toggle = GetToggle(toggleName);
-
-            var evaluationResult = CheckIsEnabled(toggleName, context, false, defaultValue);
-
-            RegisterCount(toggleName, evaluationResult.Enabled);
-
-            RegisterVariant(toggleName, evaluationResult.Variant);
-
             var enhancedContext = context.ApplyStaticFields(settings);
 
-            if (toggle?.ImpressionData ?? false)
-            {
-                EmitImpressionEvent("getVariant", enhancedContext, evaluationResult.Enabled, toggle.Name, evaluationResult.Variant?.Name);
-            }
-
-            return evaluationResult.Variant;
-        }
-
-        public IEnumerable<VariantDefinition> GetVariants(string toggleName)
-        {
-            return GetVariants(toggleName, services.ContextProvider.Context);
-        }
-
-        public IEnumerable<VariantDefinition> GetVariants(string toggleName, UnleashContext context)
-        {
-            if (!IsEnabled(toggleName, context)) return null;
-
-            var toggle = GetToggle(toggleName);
-
-            return toggle?.Variants;
-        }
-
-        private FeatureToggle GetToggle(string toggleName)
-        {
-            return services
-                .ToggleCollection
-                .Instance
-                .GetToggleByName(toggleName);
-        }
-
-        private void RegisterCount(string toggleName, bool enabled)
-        {
-            if (services.IsMetricsDisabled)
-                return;
-
-            services.MetricsBucket.RegisterCount(toggleName, enabled);
-        }
-
-        private void RegisterVariant(string toggleName, Variant variant)
-        {
-            if (services.IsMetricsDisabled)
-                return;
-
-            services.MetricsBucket.RegisterCount(toggleName, variant.Name);
-        }
-
-        private static IStrategy[] SelectStrategies(IStrategy[] strategies, bool overrideDefaultStrategies)
-        {
-            if (overrideDefaultStrategies)
+            var variant = services.engine.GetVariant(toggleName, enhancedContext) ?? defaultValue;
+            
+            services.engine.CountVariant(toggleName, variant.Name);
```

Comment:
There's some subtle stuff that may need love in the metrics for variants (I'm so sorry, the variant metrics have grown in weird ways).

- You need to count the feature yourself ala engine.CountToggle
- If the toggle doesn't exist, you should count the toggle as disabled but not the variant

(also thanks I realise there's a bug in the Ruby SDK Yggdrasil one, I count the disabled variant not the one specified by user - I think we don't support that in the UI so we probably have some work to do there as well)

Reply:
Addressed in https://github.com/Unleash/unleash-client-dotnet/pull/224/commits/63a3d68bdefa887b9fbafc6494649cb435018544